aws/offsite/lambda/verify-database-copy/index.js

```diff
@@ -0,0 +1,114 @@
+                // 2) Not Available (Status = resetting-master-credentials).  Attempt to connect as a mysql client or to invoke AWS describeDBInstances API fails
+                // 3) Available with new password
+                // Wait until about 2 minutes before this Lambda function reaches its 5 minute timeout before attempting to connect to the database
+                return setTimeoutPromise(context.getRemainingTimeInMillis() - 60000, null);
+            })
+            .then(function (value) {
+                // check if password update is pending
+                return rds.describeDBInstances({
+                        DBInstanceIdentifier: 'verification-copy'
+                    }).promise();
+            })
+            .then(function (data) {
+                if (data.DBInstances[0].PendingModifiedValues.hasOwnProperty('MasterUserPassword')) {
+                    status_message = 'Terminating verification of database copy because database password change has not started yet.';
+                    console.error(status_message);
+                    postStatusToSlack(status_message);
+                    callback(new Error(status_message));
+                } else {
+                    return mysqlPromise.createConnection({
+                        host: DB_HOST,
+                        database: DB_NAME,
+                        user: DB_USER,
+                        password: DB_PASSWORD
+                    });
+                }
+            })
+            .then(function (conn){
+                mysqlConnection = conn;
+                return mysqlConnection.query('SELECT count(*) AS number_of_users FROM users');
+            })
+            .then(function (rows){
+                status_message = 'Successfully queried offsite backup of database.  Number of Users = ' + rows[0].number_of_users;
+                console.log(status_message);
+                postStatusToSlack(status_message);
+                mysqlConnection.end();
+
+                return rds.deleteDBInstance({
+                    DBInstanceIdentifier: DB_INSTANCE_IDENTIFIER,
+                    SkipFinalSnapshot: true
+                }).promise();
+            })
+            .then(function (data){
+                callback(null, status_message);
+            })
+            .catch(function (error){
+                if (mysqlConnection && mysqlConnection.end) {
+                    mysqlConnection.end();
+                }
+                status_message = JSON.stringify(error);
+                console.log(status_message);
+                postStatusToSlack(status_message);
+                callback(error);
+            });
+    } else {
+        status_message = 'Ignore trigger because it is not one that indicates RDS Snapshot restore is complete:  ' + eventID;
+        console.log(status_message);
+        callback(null, status_message);
```

Comment:
I'd recommend changing this `else` block into an early return before the body of the `if` block to eliminate the nesting, but this is optional / personal preference.